### PR TITLE
Add option to import animations only

### PR DESCRIPTION
**Pull request by noalak**
This PR is resolving issue https://github.com/BabylonJS/Exporters/issues/537 (linked to https://github.com/BabylonJS/Exporters/issues/525).

This is the import part of the process. See PR https://github.com/BabylonJS/Exporters/pull/702 for the export part.

@Drigax, @elpie89, @Selmar are invited to test the export and import process. Samples are provided in the exporter AnimationsOnly samples folder or can be downloaded via [WeTransfer here](https://we.tl/t-ThdhgUh0dF).

The idea is to first load a scene file and then any animation file.
The node names and hierarchy must be identical to be able to link animation and scene files.

You can import animation files through:
1) The Inspector > Tools tab
2) The babylon sandbox bottom menu, once a first file has been loaded

ImportAnimationsUI

### CODE REVIEW

**Comment by noalak** [comment]
Next are TECHNICAL notes.

### USE CASES

The import animation button in bottom menu of the sandbox will perform 2 operations:
- load animations
- update the sandbox bottom bar (animation progress bar and animation group dropdown).

The import animation button in the inspector will only load animations. There are not callbacks available by the sandbox.

The use cases are expected to be:
- in the sandbox, use the sandbox button
- outside of the sandbox, use the inspector window

### CONSECUTIVE LOADS OF ANIMATIONS

The simple animations are always **discarded** between 2 consecutive loads of animation files.

The group animations are **appended** between 2 consecutive loads of animation files. An animation group with same name as one previously loaded will **override** the previous one.

Note that from the 3dsMax and Maya exporters, all animations are exported in **glTF** as group animations, even the simpliest ones.
Thus in glTF, simple animations are exported as a animation group named "All animations". Consecutive loads of such simple animations will result in discarding previous animations since the animation group name is always the same.

File: inspector/src/components/actionTabs/actionTabs.scss [hunk 1]
```diff
@@ -294,7 +294,7 @@ $line-padding-left: 2px;
                     .file-upload {
                         background: #222222;
                         border: 1px solid rgb(51, 122, 183);
-                        margin: 0px 10px;
+                        margin: 5px 10px 5px 10px;
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/actionTabs.scss]: Why this change?

File: loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts [hunk 2]
```diff
@@ -76,6 +81,17 @@ export class KHR_lights implements IGLTFLoaderExtension {
         }
     }
 
+    private _assignExtras(
```
Comment (deltakosh) [inline loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: pinging @bghgary  for review of this part

File: sandbox/index-local.html [hunk 3]
```diff
@@ -56,6 +56,12 @@
                 </a>
                 <a href="javascript:void(null);" id="btnInspector" class="hidden"><img src="./Assets/Icon_EditModel.svg"
                         alt="Display inspector" title="Display inspector" /></a>
```
Comment (deltakosh) [inline sandbox/index-local.html]: Do not update the main bar. I prefer to keep everything UI related into the inspector

File: src/Animations/animationGroup.ts [hunk 4]
```diff
@@ -575,7 +575,7 @@ export class AnimationGroup implements IDisposable {
             }
         }
 
-        if (parsedAnimationGroup.from !== null && parsedAnimationGroup.from !== null) {
+        if (parsedAnimationGroup.from !== null && parsedAnimationGroup.to !== null) {
```
Comment (deltakosh) [inline src/Animations/animationGroup.ts]: good catch :)

File: src/Loading/sceneLoader.ts [hunk 5]
```diff
@@ -976,4 +978,143 @@ export class SceneLoader {
...
+            }
+        };
+
+        // Reset, stop and dispose all animations before loading new ones
+        for (let animatable of scene.animatables) {
+            animatable.reset();
+        }
+        scene.stopAllAnimations();
+        scene.animationGroups.slice().forEach(animationGroup => {
+            // animationGroup.dispose();
```
Comment (deltakosh) [inline src/Loading/sceneLoader.ts]: comment?

File: src/Loading/sceneLoader.ts [hunk 6]
```diff
@@ -976,4 +978,143 @@ export class SceneLoader {
...
+                animatable.stop();
+            })
+        });
+
+        // Copy animatables
+        scene.animatables.slice().forEach(animatable => {
+            let target = _targetConverter(animatable.target);
+
+            // If the animatable has just been loaded
+            if (target && target != animatable.target) {
```
Comment (deltakosh) [inline src/Loading/sceneLoader.ts]:
```suggestion
            if (target && target !== animatable.target) {
```

File: src/Loading/sceneLoader.ts [hunk 7]
```diff
@@ -976,4 +978,143 @@ export class SceneLoader {
...
+
+        let _targetConverter = targetConverter ? targetConverter : this._defaultTargetConverter(scene);
+
+        let onAssetContainerLoaded = (container: AssetContainer) => {
+            SceneLoader.MergeAnimations(scene, container, _targetConverter);
+
+            container.dispose();
+
+            if (onSuccess) {
+                onSuccess(scene);
```
Comment (deltakosh) [inline src/Loading/sceneLoader.ts]: Maybe we could add a global observable on the scene,  something like: scene.onAnimationMergedObservable so the sandbox can listen to it and update accordingly

**Comment by deltakosh** [comment]
So overall:
- Please do not change sandbox UI at all (use the observable for instance to keep it in sync)
- The API should provide an option to merge instead of overwrite animations

File: src/Loading/sceneLoader.ts [hunk 8]
```diff
@@ -976,4 +978,143 @@ export class SceneLoader {
...
+            }
+        };
+
+        // Reset, stop and dispose all animations before loading new ones
```
Comment (deltakosh) [inline src/Loading/sceneLoader.ts]: This should be optional

File: src/Loading/sceneLoader.ts [hunk 9]
```diff
@@ -976,4 +978,143 @@ export class SceneLoader {
...
+        for (let animatable of scene.animatables) {
+            animatable.reset();
+        }
+        scene.stopAllAnimations();
+        scene.animationGroups.slice().forEach(animationGroup => {
+            // animationGroup.dispose();
+        });
+        let animatableObjects = this._getAllAnimatableObjects(scene);
+        animatableObjects.forEach(animatableObject => {
+            animatableObject.animations = new Array<Animation>();
```
Comment (deltakosh) [inline src/Loading/sceneLoader.ts]: Not sure to get why we need this

File: src/Loading/sceneLoader.ts (hunk 5 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: The way how to handle consecutive loads of animation files is under discussion.
Currently, the animations groups are appended, reflected in the comment of the dispose method.
We can update the PR right now to delete this comment or wait until comments are made on this behavior. As you wish.

File: src/Loading/sceneLoader.ts (hunk 7 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Ok!

File: src/Loading/sceneLoader.ts (hunk 6 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Ok

File: sandbox/index-local.html (hunk 3 above)
Comment (noalak) [reply sandbox/index-local.html]: No extra button in sandbox, but a callback when the animation scene is loaded that updates the animation group drop down and co. All right.

File: src/Loading/sceneLoader.ts (hunk 5 above)
Comment (deltakosh) [reply src/Loading/sceneLoader.ts]: if we are replacing then we should dispose the animationgroups and replace them but if we append then we should do nothing but just appending

File: loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts [hunk 10]
```diff
@@ -114,6 +130,8 @@ export class KHR_lights implements IGLTFLoaderExtension {
 
                 GLTFLoader.AddPointerMetadata(babylonLight, extensionContext);
 
+                this._assignExtras(babylonLight, node);
```
Comment (bghgary) [inline loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: I don't understand this change. The parent node will already have all the extras on it via the ExtrasAsMetadata loader extension. This seems like duplicate information?

File: src/Loading/sceneLoader.ts [hunk 11]
```diff
@@ -976,4 +978,143 @@ export class SceneLoader {
...
+        });
+    }
+
+    /**
+     * Merge animations from an asset container into a scene
+     * @param scene is the instance of BABYLON.Scene to append to (default: last created scene)
+     * @param animationAssetContainer is the instance of BABYLON.AssetContainer containing animations
+     * @param targetConverter defines a function used to convert animation targets from the asset container to the scene (default: search node by name)
+     */
+    public static MergeAnimations(scene: Scene, animationAssetContainer: AssetContainer, targetConverter: Nullable<(target: any) => any> = null): void {
```
Comment (bghgary) [inline src/Loading/sceneLoader.ts]: Is this intended to be public? If so, maybe it makes more sense to be an instance method on AssetContainer instead of a static method on SceneLoader?

File: inspector/src/components/actionTabs/actionTabs.scss (hunk 1 above)
Comment (noalak) [reply inspector/src/components/actionTabs/actionTabs.scss]: Now (before PR):

UI inspector

With the 5px vertical margin:
UI inspector new

The margins look more consistent top and bottom with the 5px.
The file input component is not often used, if not at all so that may explain why it stayed like this.

Comment (noalak) [reply inspector/src/components/actionTabs/actionTabs.scss]: This way, the margin: 5px 10px 5px 10px; is the same as the "button" one.

File: loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts (hunk 10 above)
Comment (noalak) [reply loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: Indeed. Accessing the extras/metadata on the light itself looks more intuitive to me but you certainly had such discussion before. Tell me if you want me to remove the extras from the light.

File: src/Loading/sceneLoader.ts (hunk 11 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Yes it is, in an hypothetical case where you are performing the loading manually.
Moving the method to AssetContainer seems great!

File: loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts (hunk 10 above)
Comment (bghgary) [reply loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: I think for consistency sake, it should just be on the parent node so that it matches the glTF.

Comment (noalak) [reply loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: Ok!

Comment (bghgary) [reply loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: If the light extension itself had extras, then it might make sense to do what you are doing here but for the light extension instead of the node.

File: src/Loading/sceneLoader.ts (hunk 9 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Simply a clean up of the mesh.animations, light.animations, etc when importing in overwrite mode.
The append mode will not do that of course.

File: loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts (hunk 10 above)
Comment (noalak) [reply loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: Done in https://github.com/BabylonJS/Babylon.js/pull/7252/commits/7c892e0aae5ae7606fa23989738d49c61e182b6b

File: src/Loading/sceneLoader.ts (hunk 6 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Done in https://github.com/BabylonJS/Babylon.js/pull/7252/commits/7c892e0aae5ae7606fa23989738d49c61e182b6b

File: src/Loading/sceneLoader.ts (hunk 5 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Done in https://github.com/BabylonJS/Babylon.js/pull/7252/commits/7c892e0aae5ae7606fa23989738d49c61e182b6b

File: sandbox/index-local.html (hunk 3 above)
Comment (noalak) [reply sandbox/index-local.html]: No extra button done in https://github.com/BabylonJS/Babylon.js/pull/7252/commits/7c892e0aae5ae7606fa23989738d49c61e182b6b

Callback coming next

File: src/Loading/sceneLoader.ts (hunk 8 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Ok

File: src/Loading/sceneLoader.ts (hunk 9 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: I assume we overwrite animations with same target property as the ones being loaded, even in append mode to get a clean array, right ?
Anyway, the animation classes are already handling the case where a target node has several animations with same target property.

Comment (deltakosh) [reply src/Loading/sceneLoader.ts]: > I assume we overwrite animations with same target property as the ones being loaded, even in append mode to get a clean array, right ?

Correct

File: src/Loading/sceneLoader.ts (hunk 11 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Done in https://github.com/BabylonJS/Babylon.js/pull/7252/commits/a5d84cfb691493950c138f3167bea9e2fcda1048

File: src/Loading/sceneLoader.ts (hunk 9 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Done in https://github.com/BabylonJS/Babylon.js/pull/7252/commits/bd48adfd1f20ddeec757a42990793adff611fd48

File: src/Loading/sceneLoader.ts (hunk 8 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: We have added the following parameters to the API:
- '**overwriteAnimations**' when true, animations are cleaned before importing new ones. Animations are appended otherwise
- '**animationGroupLoadingMode**' defines how to handle old animations groups before importing new ones

The resulting UI is:

2019-12-04_19h32_01

and this when appending animations:
2019-12-04_19h32_52

I recognize that the label could be better :-)

The options for **animations groups** are:
- CLEAN : Reset all old animations to initial state then dispose them.
- STOP : Stop all old animations.
- SYNC : Restart old animations from first frame.
- NOSYNC : Old animations remains untouched.

File: src/Loading/sceneLoader.ts (hunk 7 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: Done in https://github.com/BabylonJS/Babylon.js/pull/7252/commits/3130b945ece58712d89339e861b6db08af9eb5e3

File: loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts [hunk 12]
```diff
@@ -38,6 +39,10 @@ interface ILights {
     lights: ILight[];
 }
 
+interface ObjectWithMetadata {
```
Comment (deltakosh) [inline loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: why this?

Comment (noalak) [reply loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: No use anymore. Good catch!

File: sandbox/index.css [hunk 13]
```diff
@@ -129,6 +129,10 @@ a:visited {
     height: var(--footer-height);
 }
 
+.custom-upload-animation {
```
Comment (deltakosh) [inline sandbox/index.css]: not needed

File: sandbox/index.html [hunk 14]
```diff
@@ -69,8 +66,9 @@
                 </a>
                 <a href="javascript:void(null);" id="btnInspector" class="hidden"><img src="./Assets/Icon_EditModel.svg"
                         alt="Display inspector" title="Display inspector" /></a>
-                <div id="dropdownContent-env" class="hidden">
-                </div>
+                <a href="javascript:void(null);" id="animationFilesButton" class="hidden">
```
Comment (deltakosh) [inline sandbox/index.html]: not needed (I would like no change at all on sandbox)

File: src/Loading/sceneLoader.ts [hunk 15]
```diff
@@ -222,6 +223,31 @@ export interface ISceneLoaderPluginAsync extends ISceneLoaderPluginBase {
...
 }
 
+/**
+ * Mode that determines how to handle old animation groups before loading new ones.
+ */
+export enum SceneLoaderAnimationGroupLoadingMode {
+    /**
+     * Reset all old animations to initial state then dispose them.
+     */
+    CLEAN = 0,
```
Comment (deltakosh) [inline src/Loading/sceneLoader.ts]: Enum are Camel cased (Clean, Stop,etc..)

File: src/Loading/sceneLoader.ts [hunk 16]
```diff
@@ -976,4 +1002,100 @@ export class SceneLoader {
...
+            for (let animatable of scene.animatables) {
+                animatable.reset();
+            }
+            scene.stopAllAnimations();
+            scene.animationGroups.slice().forEach(animationGroup => {
+                animationGroup.dispose();
+            });
+            let nodes = scene.getNodes();
+            nodes.forEach(node => {
+                node.animations = new Array<Animation>();
```
Comment (deltakosh) [inline src/Loading/sceneLoader.ts]: this should be:
if (node.animations) {
node.animations = [];
}

File: src/assetContainer.ts [hunk 17]
```diff
@@ -454,4 +458,67 @@ export class AssetContainer extends AbstractScene {
...
+            return;
+        }
+
+        let _targetConverter = targetConverter ? targetConverter : (target: any) => { return scene.getBoneByName(target.name) || scene.getNodeByName(target.name) };
+
+        // Copy node animations
+        let nodesInAC = this.getNodes();
+        nodesInAC.forEach(nodeInAC => {
+            let nodeInScene = _targetConverter(nodeInAC);
+            if (nodeInScene != null) {
```
Comment (deltakosh) [inline src/assetContainer.ts]: !==

File: src/assetContainer.ts [hunk 18]
```diff
@@ -454,4 +458,67 @@ export class AssetContainer extends AbstractScene {
...
+            animationGroupInAC.clone(animationGroupInAC.name, _targetConverter);
+
+            // Remove animatables related to the asset container
+            animationGroupInAC.animatables.forEach(animatable => {
+                animatable.stop();
+            })
+        });
+
+        // Copy animatables
+        scene.animatables.slice().forEach(animatable => {
```
Comment (deltakosh) [inline src/assetContainer.ts]: Shouldn't that just happen for NEW animations and not all (including the ones already there) ?

**Comment by deltakosh** [comment]
You must run npm run build anyway to make sure the PR is correct :) That will update the d.ts for you but yeah please ignore the dist files for the PR itself

File: inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx [hunk 19]
```diff
@@ -173,33 +217,44 @@ export class ToolsTabComponent extends PaneComponent {
...
                 </LineContainerComponent>
+                <LineContainerComponent globalState={this.props.globalState} title="SCENE IMPORT">
+                    <FileMultipleButtonLineComponent label="Import animations" accept="gltf" onClick={(evt: any) => this.importAnimations(evt)} />
+                    <CheckBoxLineComponent label="Overwrite animations" target={sceneImportDefaults} propertyName="overwriteAnimations" onSelect={value => {
+                        sceneImportDefaults["overwriteAnimations"] = value;
+                        this.forceUpdate();
+                    }} />
+                    {
+                        sceneImportDefaults["overwriteAnimations"] === false &&
+                        <OptionsLineComponent label="Old animation groups loading mode" options={animationGroupLoadingModes} target={sceneImportDefaults} propertyName="animationGroupLoadingMode" />
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx]: Let's call it "Merge mode"

File: sandbox/animation.js [hunk 20]
```diff
@@ -45,8 +45,8 @@ dropdownBtn.addEventListener("click", function() {
 });
 
 function selectCurrentGroup(group, index, animation) {
-    if (currentGroupIndex !== undefined) {
-        document.getElementById(formatId(currentGroup.name + "-" + currentGroupIndex)).classList.remove("active");
+    for (var i = 0; i < dropdownContent.children.length; i++) {
```
Comment (deltakosh) [inline sandbox/animation.js]: why?

File: src/assetContainer.ts (hunk 18 above)
Comment (noalak) [reply src/assetContainer.ts]: The next if statement is making sure to hit only new animations

`if (target && target !== animatable.target) {`

File: inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx (hunk 19 above)
Comment (noalak) [reply inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx]: Ok i wanted to highlight the fact this option is only targeting animation **groups**

File: src/assetContainer.ts (hunk 18 above)
Comment (deltakosh) [reply src/assetContainer.ts]: Well I would argue that this will not be required if you used imported animatables list instead of the entire scene.animatables

File: loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts (hunk 12 above)
Comment (noalak) [reply loaders/src/glTF/2.0/Extensions/KHR_lights_punctual.ts]: done

File: sandbox/index.css (hunk 13 above)
Comment (noalak) [reply sandbox/index.css]: done

File: sandbox/index.html (hunk 14 above)
Comment (noalak) [reply sandbox/index.html]: done

File: src/Loading/sceneLoader.ts (hunk 15 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: done

File: src/Loading/sceneLoader.ts (hunk 16 above)
Comment (noalak) [reply src/Loading/sceneLoader.ts]: done

File: src/assetContainer.ts (hunk 17 above)
Comment (noalak) [reply src/assetContainer.ts]: done

File: src/assetContainer.ts (hunk 18 above)
Comment (noalak) [reply src/assetContainer.ts]: New animatables from assetContainer are stored directly into the scene.animatables array when autostarting node animations.
I can store the array before that and compare with updated one after animation file import.
The current scene.animatables array would be a parameter of the MergeAnimationsTo method.
Or use the array length as index for the new animatables.

File: inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx (hunk 19 above)
Comment (noalak) [reply inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx]: done

File: sandbox/animation.js (hunk 20 above)
Comment (noalak) [reply sandbox/animation.js]: i had a bug on this but can't reproduce it now.
I revert to the update of current option.

File: src/assetContainer.ts (hunk 18 above)
Comment (noalak) [reply src/assetContainer.ts]: done

File: sandbox/animation.js (hunk 20 above)
Comment (noalak) [reply sandbox/animation.js]: done

**Comment by deltakosh** [comment]
Will now you need to check why the CI is failing :D

**Comment by noalak** [comment]
I still need to build successfully on my PC :)

**Comment by deltakosh** [comment]
Good luck ;D

File: dist/preview release/what's new.md [hunk 21]
```diff
@@ -92,6 +93,7 @@
 - Added diff navigator in the playground ([sailro](http://www.github.com/sailro))
 - Added custom filter to remove internals from the completion in the playground ([sailro](http://www.github.com/sailro))
 - Added support for tagging deprecated members (both in editor and for completion) in the playground ([sailro](http://www.github.com/sailro))
+- Updated the sandbox to autoplay the last animation group loaded instead of the first one ([noalak](https://github.com/noalak/))
```
Comment (deltakosh) [inline dist/preview release/what's new.md]: We still need to play the first one when loading a regular gltf for instance

File: inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx [hunk 22]
```diff
@@ -34,6 +38,14 @@ export class ToolsTabComponent extends PaneComponent {
         super(props);
 
         this.state = { tag: "Record video" };
+
+        const sceneImportDefaults = this.props.globalState.sceneImportDefaults;
+        if (sceneImportDefaults["overwriteAnimations"] === undefined) {
+            sceneImportDefaults["overwriteAnimations"] = true;
+        }
+        if (sceneImportDefaults["animationGroupLoadingMode"] === undefined) {
+            sceneImportDefaults["animationGroupLoadingMode"] = 0; // TODO - import SceneLoaderAnimationGroupLoadingMode
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx]: Need to be fixed

File: inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx [hunk 23]
```diff
@@ -163,6 +199,15 @@ export class ToolsTabComponent extends PaneComponent {
             return null;
         }
 
+        const sceneImportDefaults = this.props.globalState.sceneImportDefaults;
+
+        var animationGroupLoadingModes = [
+            { label: "Clean", value: 0 }, // TODO - import SceneLoaderAnimationGroupLoadingMode
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx]: same

File: dist/preview release/what's new.md (hunk 21 above)
Comment (noalak) [reply dist/preview release/what's new.md]: Ok

Comment (noalak) [reply dist/preview release/what's new.md]: done

File: inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx (hunk 22 above)
Comment (noalak) [reply inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx]: done

File: inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx (hunk 23 above)
Comment (noalak) [reply inspector/src/components/actionTabs/tabs/toolsTabComponent.tsx]: done

**Comment by noalak** [comment]
Build -> successful!
TODO -> done!

**Comment by deltakosh** [comment]
Yeah Congrats!!! Do you mind updating the doc and provide some examples files for the doc?
https://doc.babylonjs.com/how_to/load_from_any_file_type

**Comment by noalak** [comment]
Sure
